hooks/pre-commit

```diff
@@ -1,25 +1,27 @@
-#!/bin/bash
-if clang-format --version | grep -q 'version 11\.'; then
-   CLANG_FORMAT_EXECUTABLE="clang-format"
-else
-   CLANG_FORMAT_EXECUTABLE="clang-format-11"
-fi
+#!/bin/sh
+
+name=""
+version="14.0.0"
 
-if ! command -v $CLANG_FORMAT_EXECUTABLE &> /dev/null
-then
-    echo $CLANG_FORMAT_EXECUTABLE does not exist, make sure to install it
-    exit 1
+for file in $(find $(echo "$PATH" | sed 's/:/ /g') -maxdepth 1 -type f -executable -name 'git-clang-format*'); do
+  cur_name="$(basename "$file" | sed 's/^git-//')"
```

Comment:
Is there any difference between `git clang-format` and `git-clang-format`? If not, we can remove this step.

Reply:
This is in there on the off-chance that a clang-format installation doesn't include `git-clang-format`. The reason for removing the `git-` prefix is because `git-clang-format` itself doesn't have a way to print the version, so we have to use `clang-format --version`. If you think it's safe to assume every clang-format installation has an accompanying `git-clang-format`, we can very slightly simplify this code.

Reply:
Well this script depends on `git-clang-format`, but regarding this I think I'm just confusing the different commands.